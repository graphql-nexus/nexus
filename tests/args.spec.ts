import { graphql } from "graphql";
import {
  arg,
  booleanArg,
  floatArg,
  idArg,
  intArg,
  makeSchema,
  objectType,
  queryField,
  stringArg,
} from "../src/core";

describe("interfaceType", () => {
  let schema: ReturnType<typeof makeSchema>;
  beforeAll(() => {
    schema = makeSchema({
      types: [
        queryField("user", {
          type: "User",
          args: {
            int: intArg(),
            bool: booleanArg(),
            float: floatArg(),
            id: idArg(),
            str: stringArg(),
          },
          resolve: () => ({ id: `User:1`, name: "Test User" }),
        }),
        objectType({
          name: "User",
          definition(t) {
            t.id("id");
            t.string("name");
          },
        }),
      ],
      outputs: false,
      shouldGenerateArtifacts: false,
    });
  });
  it("can be implemented by object types", async () => {
    expect(
      await graphql(
        schema,
        `
          {
            user(int: 1, bool: true, float: 123.45, str: "Test") {
              id
              name
            }
          }
        `
      )
    ).toMatchSnapshot();
  });
  it("throws if the arg is not provided to the type", async () => {
    // @ts-ignore
    expect(() => arg({ type: null })).toThrowErrorMatchingSnapshot();
  });
});
