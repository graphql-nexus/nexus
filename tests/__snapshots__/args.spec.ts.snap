// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interfaceType can be implemented by object types 1`] = `
Object {
  "data": Object {
    "user": Object {
      "id": "User:1",
      "name": "Test User",
    },
  },
}
`;

exports[`interfaceType throws if the arg is not provided to the type 1`] = `"You must provide a \\"type\\" for the arg()"`;
